### DISCUSSION TITLE
Set GameOption

### DISCUSSION THREAD

**Discussion by Grayseon** [FabricMC/fabric-api#1956] [Mod Dev Support] [answered]
Hi. I want to set the sneak toggled option to true. This gives an incorrect answer:
```new GameOptions(MinecraftClient.getInstance(), new File(String.valueOf(FabricLoader.getInstance().getGameDir().resolve("options.txt")))).sneakToggled```
and I can't set the value using that. Can I have some help, please?

Thank you!

**Comment by Player3324** [comment, marked as answer]
You shouldn't instantiate the option yourself, that wouldn't affect the game directly.

Something like `Option.SNEAK_TOGGLED.set(MinecraftClient.getInstance().options, true)` may work.

**Reply by Grayseon** [reply]
Hi. Thank you for the fast reply! .set doesn't exist in Option.SNEAK_TOGGLED sadly.

**Reply by Player3324** [reply]
Right it is a cycling option, the non-access-widener/mixin/reflection option is then unfortunately some form of 
```java
if (!MinecraftClient.getInstance().options.sneakToggled) {
  Option.SNEAK_TOGGLED.createButton(MinecraftClient.getInstance().options, 0, 0, 0).onClick(0, 0);
}
```

**Reply by Player3324** [reply]
Or just `MinecraftClient.getInstance().options.sneakToggled = true`, i think it has no side effects unlike another options where i had to go through the button

**Reply by Grayseon** [reply]
Is it bad to be doing this on startup? I have this code in the startup file, but when I do that it gives an error.
```Cannot assign field "sneakToggled" because "net.minecraft.client.MinecraftClient.getInstance().options" is null```

**Reply by Player3324** [reply]
Yes our hook for the mod entry points is obviously a little before mc initializes the options field in its constructor. Something like `ClientLifecycleEvents.CLIENT_STARTED` should be fine.

**Reply by Grayseon** [reply]
Thank you so much!
